### ISSUE TITLE
USWDS - Bug: Focus trap treats hidden inputs as focusable

### ISSUE THREAD

**Issue by sanason** [uswds/uswds#6056]
### Describe the bug

Hidden input elements are not focusable but the focus trap considers them to be.

When a hidden input is the first "focusable" element in a modal, it breaks tab cycling. Once the user has tabbed to the last element in the modal, focus gets stuck there. For example, if you're following the USWDS advice to put the close button at the end of the modal code, then focus gets stuck on the close button.

### Steps to reproduce the bug

1. Go to https://codepen.io/sanason/pen/WNqmYrg.
2. Click "Open modal".
3. Press Tab 4 times.

### Expected Behavior

On the fourth tab press, focus returns to the "Tell us what you thought:" text input, instead of remaining on the X close button.

### Additional context

Touchpoints recently started using the USWDS modal which means a lot of forms inside modals. Touchpoints forms used to begin with several hidden inputs, so this issue was fairly visible. We've since moved the hidden inputs to the end of the form, which is an easy workaround.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mahoneycm** [comment]
Hey there @sanason ! Is this still an issue for you and your team?

I was unable to reproduce the issue in the linked code pen or by using the same markup in a [Sandbox](https://github.com/uswds/uswds-sandbox) environment. 

I tested on Chrome, Safari, and Firefox and the focus trap remained on the close button.

**Comment by mahoneycm** [comment]
**Update:** I noticed that this behavior also happens on our [Nested Forms test story](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-modal--test-nested-forms&viewMode=story). 

This means the interaction is not limited to the `hidden` attribute. We should reconsider the weight of this issue as more modals may be affected.

**Comment by mejiaj** [comment]
Issue has been confirmed. We need additional refinement to understand the problem and severity.

**Comment by mejiaj** [comment]
@RachelCorsino can you look into this issue of focus trap breaking in certain cases and refine the issue based on your findings?

We we need to know:
1. If this issue is scoped properly, i.e. _is it solely a focus trap issue?_
2. What the severity of the issue is
3. An estimate of the LOE to fix

Once known please, update the issue metadata and reach out to me for potential scheduling.

**Comment by dct2023** [comment]
@RachelCorsino Following up on this. Should we move this to an upcoming sprint? Is this tied to an upcoming deliverable?

**Comment by mejiaj** [comment]
@dct2023 bumping this to `2025.3` because it's not tied to a deliverable and availability.
